docs/guide/topics/sharding.rst

```diff
@@ -0,0 +1,81 @@
+.. currentmodule:: discord
+
+.. _guide_sharding:
+
+Sharding
+==========
+
+When bots start to get considerably large, the amount of events they have to deal with can start to become problematic.
+
+At high user and guild counts the incoming messages, typing events, and status updates can start to climb to being as frequent as hundreds per second.
+
+To help deal with this large amount of traffic, Discord supports **sharding**, a feature where your bot can split its guilds amongst separate connections, reducing the amount of data each individual connection has to handle.
+
+This not only helps Discord by reducing how much data they need to direct to one place, but also helps us, as we can split our bot's overall work across different connections, environments, or even entirely different machines. 
+
+Discord recommends using sharding once you get over 1,000 guilds, and once you reach beyond 2,500 guilds, it becomes a requirement.
+
+Now, let's discuss what options we have for setting up sharding within discord.py.
```

Comment:
The sentences in this introduction section are a bit long. This makes the paragraph harder to comprehend, so it may be helpful if the sentences are broken up.